spec/validators/ukprn_validator_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe UkprnValidator do
+  subject do
+    Class.new {
+      include ActiveModel::Validations
+      attr_accessor :ukprn_value
+      validates :ukprn_value, ukprn: true
+    }.new
+  end
+
+  context "with a valid UKPRN" do
+    it "is invalid" do
+      subject.ukprn_value = 12345678
+      expect(subject.valid?).to be true
+    end
+  end
+
+  context "with a UKPRN that is longer than eight digits" do
+    it "is invalid" do
+      subject.ukprn_value = 123456789101112
```

Comment:
Should this test the smallest change that can be invalid `123456789`. (i.e., one number more than is allowed)

I can't find a link to back me up here, but I'm sure this is a thing because we can't test every possibility

Reply:
I am not sure, I don't mind as long as it is visibly not valid! Happy to change it. 👍